core/src/agent/actions/commit.rs

```diff
@@ -33,13 +33,13 @@ pub struct CommitFuture {
 }
 
 impl Future for CommitFuture {
-    type Item = ActionResponse;
-    type Error = String;
+    type Item = Address;
+    type Error = HolochainError;
```

Comment:
Added `HolochainError::ValidationFailed` below. Much clearer interface for `CommitFuture`